src/main/java/org/opensearch/knn/index/util/Faiss.java

```diff
@@ -305,7 +305,7 @@ public class Faiss extends NativeLibrary {
                     return ((4L * centroids * dimension) / BYTES_PER_KILOBYTES) + 1;
                 })
                 .build()
-        ).addSpaces(SpaceType.UNDEFINED, SpaceType.L2, SpaceType.INNER_PRODUCT).build()
+        ).addSpaces(SpaceType.UNDEFINED, SpaceType.L2, SpaceType.INNER_PRODUCT, SpaceType.HAMMING_BIT).build()
```

Comment:
Just for my understanding, is hamming bit related to binary support?

Reply:
Yes, currently in Faiss, it only support using HAMMING_BIT space type for binary format